curvefs/src/mds/fs_manager.cpp

```diff
@@ -99,8 +99,25 @@ FSStatusCode FsManager::CreateFs(const std::string& fsName,
     uint32_t gid = 0;                 // TODO(cw123)
     uint32_t mode = S_IFDIR | 01777;  // TODO(cw123)
 
-    auto ret =
-        metaserverClient_->CreateRootInode(wrapper.GetFsId(), uid, gid, mode);
+    // create partition
+    auto ret = FSStatusCode::OK;
+    curvefs::mds::topology::CreatePartitionRequest request;
+    curvefs::mds::topology::CreatePartitionResponse response;
+    request.set_fsid(wrapper.GetFsId());
+    request.set_count(1);
+    topoManager_->CreatePartition(&request, &response);
+    if (TopoStatusCode::TOPO_OK != response.statuscode() ||
+        response.partitioninfolist_size() < 1) {
+        LOG(ERROR) << "CreateFs fail, create partition fail"
+                   << ", request = " << request.ShortDebugString()
+                   << ", error code = " << response.statuscode();
+        ret = FSStatusCode::CREATE_PARTITION_ERROR;
+    } else {
+        PartitionInfo partition = response.partitioninfolist(0);
+        ret = metaserverClient_->CreateRootInode(wrapper.GetFsId(),
```

Comment:
another interface
    
// for persistence consider
    ret = fsStorage_->Update(wrapper);
    if (ret != FSStatusCode::OK) {
        LOG(ERROR) << "CreateFs fail, update fs to inited fail"
                   << ", fsName = " << fsName
                   << ", ret = " << FSStatusCode_Name(ret);
        ret = metaserverClient_->DeleteInode(wrapper.GetFsId(), GetRootId());
        if (ret != FSStatusCode::OK) {
            LOG(ERROR) << "CreateFs fail, update fs status to inited fail"
                       << ", then delete root inode fail"
                       << ", fsName = " << fsName
                       << ", ret = " << FSStatusCode_Name(ret);
            return ret;
        }